test/integration/dockerregistryclient_test.go

```diff
@@ -3,12 +3,54 @@
 package integration
 
 import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
 	"reflect"
 	"testing"
+	"time"
 
 	"github.com/openshift/origin/pkg/dockerregistry"
 )
 
+// isRegistryReachable returns false if a Get on registry url fails with similar error:
+//   Get https://registry.com/v2/: dial tcp registry.com:443: i/o timeout
+//   Get https://registry.com/v2/: dial tcp: lookup registry.com: no such host
+//   Get https://registry.com/v2/: dial tcp registry.com:443: getsockopt: connection refused
+//   Get https://registry.com/v2/: read tcp 127.0.0.1:39849->registry.com:443: read: connection reset by peer
+func isRegistryReachable(registry, version string, allowInsecure bool) (bool, string) {
+	client := http.Client{Timeout: time.Second * 4}
+	_, err := client.Get(fmt.Sprintf("https://%s/%s/", registry, version))
+	if err != nil && allowInsecure {
+		_, err = http.Get(fmt.Sprintf("http://%s/%s/", registry, version))
```

Comment:
Docker and OpenShift clients do the same. They first try `https` and if they fail, they fallback to `http`.